microsoft/testsuites/mdatp/check-mdatp.sh

```diff
@@ -0,0 +1,113 @@
+#! /bin/sh
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+# check the following folders for printable files, print them
+# checks Azure Extension directories using globs, since they're not 
+# well-known.
+
+# Returns nonzero value on detection of any defender info
+# Return value is a bitmask showing the types of info detected.
+# Bits are ordered by seriousness of the issue:
+# Least significant bit is bad
+# Most significant bit is worst.
+EXIT_CODE=0
+# if any mdatp/defender installation dir is found
+EXIT_MDATP_INSTALLED=1
+# if any log dirs are found
+EXIT_MDATP_LOGS_FOUND=2
+# if an installation log is found
+EXIT_MDATP_INSTALL_LOGS_FOUND=4
+# if an onboarding blog is found
+EXIT_ONBOARD_INFO_FOUND=8
+MDATP_OPT_DIR='/etc/opt/microsoft/mdatp'
+MDATP_LOG_DIR='/var/log/microsoft/mdatp'
+ERROR_MSG_HEADER="----------------------------------------------------------------------"
+
+find_printable_files () {
+    found_dir="$1"
+    sudo find "$found_dir" -type f -exec file '{}' ';' -exec grep -Iq . '{}' ';' -exec head -c 1024 '{}' ';'
+}
+
+check_unexpected_file () {
+    surprise_file="$1"
+    echo "Found unexpected regular file: $surprise_file" >&2
+    # see what it is, print if it's printable
+    file "$surprise_file"; grep -Iq . "$surprise_file" && cat "$surprise_file"
+}
+# check for mdatp extension, folder is version labeled, so use a shell glob
+for mde_dir in /var/lib/waagent/Microsoft.Azure.AzureDefenderForServers.MDE.Linux* ; do
+    # for all the versioned folders we find...
+    if [ -e "$mde_dir" ]; then
+        EXIT_CODE=$EXIT_MDATP_INSTALLED
+        if [ -d "$mde_dir" ]; then
+            # find regular files, skip printing them if they are binary
+            find_printable_files "$mde_dir"
+        else
+            # not expecting to find a regular file in /var/log/azure, but...
+            check_unexpected_file "$mde_dir"
+        fi
+    fi
+done
+
+# check for azure extension install dir
+if [  -e "$MDATP_OPT_DIR" ]; then
+    EXIT_CODE=$EXIT_MDATP_INSTALLED
+    find_printable_files "$MDATP_OPT_DIR"
+fi
+
+# check for install or runtime logs
+if [  -e "$MDATP_LOG_DIR" ]; then
+    EXIT_CODE=$((EXIT_CODE|EXIT_MDATP_LOGS_FOUND))
+    find_printable_files "$MDATP_LOG_DIR"
+fi
+
+# check for Azure Resource Manager mde extension logs
+for log_dir in /var/log/azure/Microsoft.Azure.AzureDefenderForServers.MDE.Linux* ; do
+    # for all the versioned folders we find...
+    if [ -e "$log_dir" ]; then
+        EXIT_CODE=$((EXIT_CODE|EXIT_MDATP_LOGS_FOUND))
+        echo "checking $log_dir..."
+        if [ -d "$log_dir" ]; then
+            find_printable_files "$log_dir"
+        else
+            # not expecting to find a regular file in /var/log/azure, but...
+            check_unexpected_file "$log_dir"
+        fi
+    fi
+done
+
+# check for ARC mde extension logs
+for log_dir in /var/lib/GuestConfig/extension_logs/Microsoft.Azure.AzureDefenderForServers.MDE.Linux-* ; do
```

Comment:
Note that the log folders on **ARC** look like:
/var/lib/GuestConfig/extension_logs/Microsoft.Azure.AzureDefenderForServers.MDE.Linux-1.0.3.12
/var/lib/GuestConfig/extension_logs/Microsoft.Azure.AzureDefenderForServers.MDE.Linux-1.0.3.13

and in **ARM** like:
/var/log/azure/Microsoft.Azure.AzureDefenderForServers.MDE.Linux 
without version's folder, just files.

Reply:
Thank you! I've made each section more clear and added an exit code to differentiate the extension install from the agent install. The test fails either way but we'll get a nice error message detailing what was present.